### PR TITLE
Add number entity support

### PR DESCRIPTION
**Pull request by Shulyaka**
## Breaking change

N/A

## Proposed change

Add frontend support for the new `number` entity.

number entities

Currently it is shown as a box because it is the default for the `input_number` which we are re-using. But I will raise a PR to core to allow slider mode, which will be supported automatically.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
demo:
```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion: home-assistant/core#42735
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by thomasloven** [comment]
Does this consider the more-info dialog?

**Comment by Shulyaka** [comment]
There is no more-info dialog, which is same as with input_number.

File: src/common/const.ts [hunk 1]
```diff
@@ -77,6 +78,7 @@ export const DOMAINS_WITH_CARD = [
   "input_text",
   "lock",
   "media_player",
+  "number",
```
Comment (bramkragten) [inline src/common/const.ts:81]: This means you would need to add a specific state card for this domain, like the `input_number` also has:
<https://github.com/home-assistant/frontend/blob/dev/src/state-summary/state-card-input_number.js#L26>

**Comment by thomasloven** [comment]
There is a more-info dialog for input_number. I believe it uses the state card from the states interface in some way 
image

**Comment by bramkragten** [comment]
Yes, that is the state card I linked above.

**Comment by Shulyaka** [comment]
Ok, got it. Is `stateCardType()` the correct way to do it (please see last commit), or I'd better inherit on `StateCardInputNumber`?

**Comment by bramkragten** [comment]
> Ok, got it. Is `stateCardType()` the correct way to do it (please see last commit), or I'd better inherit on `StateCardInputNumber`?

It is pretty dangerous to use the UI of Input Number everywhere, it means we always have to check if the 2 are in sync when changes are made.

I think we should just copy them

**Comment by Shulyaka** [comment]
> It is pretty dangerous to use the UI of Input Number everywhere, it means we always have to check if the 2 are in sync when changes are made.
> 
> I think we should just copy them

Copied `state-card-input_number`. Should I copy `hui-input-number-entity-row` as well?

**Comment by bramkragten** [comment]
Yes please :-)

**Comment by Shulyaka** [comment]
Done.
But now when we fix one of them, we must remember to fix the other.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**